### PR TITLE
chore(ci): run file uploader tests on fileuploader/main release

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes

This change adds tests to run on the `fileuploader/main` release branch.

Needs to be merged before #2880

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/test-fileuploader-release-prs.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if: always() && github.event.label.name == 'run-tests'
+        uses: actions/github-script@0.3.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { issue: { number: issue_number }, repo: { owner, repo } } = context;
+            const label = 'run-tests';
+            github.issues.removeLabel({ owner, repo, issue_number, name: label });
+  test:
+    uses: aws-amplify/amplify-ui/.github/workflows/reusable-e2e.yml@fileuploader/main
```
Comment (ErikCH) [inline .github/workflows/test-fileuploader-release-prs.yml:43]: uses the `reusable-e2e` tests on `fileuploader/main`.

**Review by wlee221 — COMMENTED** [review]
Looking good, let's wait til we add `AWS_ACCESS_KEY_ID_STORAGE`

**Comment by ErikCH** [comment]
>

File: .github/workflows/test-fileuploader-release-prs.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    needs: setup
+    with:
+      commit: ${{ github.event.pull_request.head.sha }}
+      repository: ${{ github.event.pull_request.head.repo.full_name }}
+    secrets:
+      # TODO: migrate away from this account once tfn is migrated
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      # TODO: migrate away from this account once tfn is migrated
```
Comment (dbanksdesign) [inline .github/workflows/test-fileuploader-release-prs.yml:51]: We can probably remove this comment since it is duplicated?

**Review by dbanksdesign — APPROVED** [review]
Just one minor comment, but not blocking.

File: .github/workflows/test-fileuploader-release-prs.yml (hunk 2 above)
Comment (wlee221) [reply .github/workflows/test-fileuploader-release-prs.yml:51]: IMO it's okay if we leave it; the todo refers to the first two secrets there so I think having it duplicated makes it clear

**Review by wlee221 — APPROVED** [review]

**Pull request merged**